client/src/mods/Character.lua

```diff
@@ -348,6 +348,50 @@ function Character.graphics_uninit(self)
   self.telegraph_garbage_images = {}
 end
 
+function Character:initializeStackAnimation(stack)
+  local character = characters[stack.character]
+  if character.battleAnimations then
+    stack.battleAnimations = {}
+    for name, set in pairs(character.battleAnimations) do
+      local anim = set:clone()
+      anim:stop()
+      anim:setSwitchFunction(
+        function (s, state)
+          local switch
+          local finish = false
+          if stack.match.ended then
+            switch = (stack.game_over_clock <= 0 and "win") or "lose"
+          elseif state == "hurt" and stack.shake_time > 0 then
+            switch = "hurt"
+          elseif state == "attack" then
+            switch = "attack"
+          elseif (stack.danger)then
+            switch = "danger"
+            finish = anim.currentAnim ~= "normal"
+          else
+            switch = "normal"
+            finish = true
+          end
+          if anim.animations[switch] ~= nil then
+            s:switchAnimation(switch, finish)
+          end
+        end
+      )
+      local landFunc = function(animation, stack, shake)
+        animation:switchAnimation("hurt", false)
+      end
+      local matchedFunc = function(animation, stack, attackGfxOrigin, isChainLink, comboSize, metalCount, garbagePanelCount)
+        if isChainLink or comboSize > 3 or metalCount > 1 then
+          animation:switchAnimation("attack", false)
+        end
+      end
+      stack.battleAnimations[name] = anim
+      stack:connectSignal("garbageLanded", stack.battleAnimations[name], landFunc)
+      stack:connectSignal("matched", stack.battleAnimations[name], matchedFunc)
```

Comment:
As the new signals are more generic, I added these two functions as glue to connect the generic signal to what the current implementation for battle sprites wants without having to touch the latter or compromise on my goals for the stack signals.
This may see some more changes later on but for now it seems to work fine.

Reply:
Oh yeah that does make a lot of sense. I didn't think of that before so when I tried this it caused both sides to be the same. I'm still kinda learning how best to handle variable scope with Lua but this should work yeah.